prometheus: avoid preallocating scope attributes when disabled (#8404)
open-telemetry/opentelemetry-go: CHANGELOG.md

### Fixed

> - Avoid preallocating scope attributes in Prometheus exporter when disabled. (#8404)

Reviewer: ```suggestion
- Avoid preallocating scope attributes when they are disabled in `go.opentelemetry.io/otel/exporters/prometheus` . (#8404)
```
